### PR TITLE
update lock file to mitigate critical vulnerability in httpx

### PR DESCRIPTION
**Pull request by skonapur**

### CODE REVIEW

**Comment by sprohaska-vouch** [comment]
Hello! Following up on this. You mentioned that this was a dependency elsewhere - by that you meant Mode, right?

**Comment by sprohaska-vouch** [comment]
So long as it's passing [tests](https://github.com/svinstech/mode-client/blob/main/tests/test_clients.py), I'm not super concerned about the potential impact of underlying deps changing, especially when weighed against security concerns about leaving these unpatched.

If we do find any issues not handled by the test cases, I'll cut tickets to update tests accordingly. Mode is sort of an odd edge case for the DE team, as we aren't directly responsible for it (mostly) but do have some utils like this floating around.

I'll check out the downstream deps and update here.

**Comment by sprohaska-vouch** [comment]
Downstream dependencies surfaced via this [GH search](https://github.com/search?q=org%3Asvinstech+mode-client+NOT+repo%3Asvinstech%2Fmode-client&type=code):
- [x] [`mode-bot`](https://github.com/svinstech/mode-bot): Archived/deprecated. Blast radius nil.
- [x] [`tap-anvil`](https://github.com/svinstech/tap-anvil/blob/4dad2d23af95ef0cd59c79ee8e4d6decdabb6104/README.md?plain=1#L18): I'm 99% sure this is actually a typo. Doesn't make sense in context + isn't in the [pyproject](https://github.com/svinstech/tap-anvil/blob/4dad2d23af95ef0cd59c79ee8e4d6decdabb6104/pyproject.toml). Blast radius so small it rounds to 0.
- [ ] [`mode-dashboards`](https://github.com/svinstech/mode-dashboards/blob/76b35428d54366359a2ed4738bfc386459b196a0/pyproject.toml#L10): Breaking this would have a severe impact on the analytics team, and prevent their delivery of work product to teams downstream. Blast radius: ⚠️ Moderate, exercise caution.


It's worth noting that the tests I linked above aren't actually running in CI (mostly because there is no CI for this repo). I'm not sure we touch this repo enough to merit building that out now, but... it'd be nice. Noting this in the README and asking PR submitters to run tests before merge would be a good start.

TLDR: If one of us can run the test on our machine, so long as they pass I have no issue with merging this branch back in. Once merged I'll keep an eye out for issues with the dashboard deploy; I'm on call this week so would be a focus for me anyway.

**Comment by skonapur** [comment]
Hey @sprohaska-vouch, I tried to run the tests locally. I get a `ModuleNotFoundError: No module named 'mode_client'`. I feel like I am missing something basic. Can you point me in the right direction?
